x-pack/filebeat/input/httpjson/request.go

```diff
@@ -96,24 +100,44 @@ type requestFactory struct {
 	password   string
 	log        *logp.Logger
 	encoder    encoderFunc
+	replace    string
+	split      *split
 }
 
-func newRequestFactory(config *requestConfig, authConfig *authConfig, log *logp.Logger) *requestFactory {
+func newRequestFactory(config config, log *logp.Logger) []*requestFactory {
 	// config validation already checked for errors here
-	ts, _ := newBasicTransformsFromConfig(config.Transforms, requestNamespace, log)
+	var rfs []*requestFactory
+	ts, _ := newBasicTransformsFromConfig(config.Request.Transforms, requestNamespace, log)
+	// regular call requestFactory object
 	rf := &requestFactory{
-		url:        *config.URL.URL,
-		method:     config.Method,
-		body:       config.Body,
+		url:        *config.Request.URL.URL,
+		method:     config.Request.Method,
+		body:       config.Request.Body,
 		transforms: ts,
 		log:        log,
-		encoder:    registeredEncoders[config.EncodeAs],
+		encoder:    registeredEncoders[config.Request.EncodeAs],
+	}
+	if config.Auth != nil && config.Auth.Basic.isEnabled() {
+		rf.user = config.Auth.Basic.User
+		rf.password = config.Auth.Basic.Password
 	}
-	if authConfig != nil && authConfig.Basic.isEnabled() {
-		rf.user = authConfig.Basic.User
-		rf.password = authConfig.Basic.Password
+	rfs = append(rfs, rf)
+	for _, ch := range config.Chain {
```

Comment:
This terminates after the first level of the nesting of the `Chain` field. The original design considered the full n-ary tree of chains which would require a recursive construction here rather than a flattening into `rfs`. The power gained by that may not be necessary (it would allow bounded conditional execution, which was the design motivation at the time), but if it is, that construction is needed. If not, there should be documentation that notes that deeper nesting is ignored.

Reply:
Yes @efd6 , I remember this was one of the implementation proposals that was discussed during our design meeting. We discussed it with the team and to make the implementation simpler, we thought of using the current approach. And I agree with your point, we should definitely call this out in the documentation that nesting is not allowed to avoid any confusion.